src/dev/flang/parser/Lexer.java

```diff
@@ -1198,7 +1198,9 @@ public SourceRange sourceRange(int prevPos, int pos, int endPos)
        endPos  <= byteLength());
 
     return pos < endPos ? sourceRange(pos, endPos)
-                        : sourceRange(prevPos, prevPos+1);
+                        : prevPos+1 <= byteLength()
+                        ? sourceRange(prevPos, prevPos+1)
+                        : null;
```

Comment:
Can you instead do
```suggestion
                        : sourceRange(prevPos, Math.min(byteLength(), prevPos+1));
```
Such that the position is not fully lost in this case?